x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,366 @@
+        }),
+        message: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.message', {
+          defaultMessage:
+            '"Native Connectors" are only intended for Elastic-managed environments like Elastic Cloud and ' +
+            'Elastic Serverless. Any connectors with `is_native: true` must be converted to connector clients or deleted ' +
+            'before this upgrade can proceed.',
+        }),
+        documentationUrl:
+          'https://elastic.co/guide/en/enterprise-search/current/upgrading-to-9-x.html',
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.listConnectors', {
+              defaultMessage: 'Enumerate all connector records',
+            }),
+            i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.convertPretenders', {
+              defaultMessage: '"Convert to Client" any that have `is_native: true`.',
+            }),
+          ],
+          api: {
+            method: 'POST',
+            path: '/internal/enterprise_search/deprecations/convert_connectors_to_client',
+            body: {
+              ids: nativeConnectors.map((it) => it.id),
+            },
+          },
+        },
+      });
+    } else {
+      const nativeServiceTypes = [
+        'azure_blob_storage',
+        'box',
+        'confluence',
+        'dropbox',
+        'github',
+        'gmail',
+        'google_cloud_storage',
+        'google_drive',
+        'jira',
+        'mssql',
+        'mongodb',
+        'mysql',
+        'network_drive',
+        'notion',
+        'onedrive',
+        'oracle',
+        'outlook',
+        'postgresql',
+        's3',
+        'salesforce',
+        'servicenow',
+        'sharepoint_online',
+        'sharepoint_server',
+        'slack',
+        'microsoft_teams',
+        'zoom',
+      ];
+      const nativeTypesStr = '[' + nativeServiceTypes.join(', ') + ']';
+      const fauxNativeConnectors = nativeConnectors.filter(
+        (hit) => !nativeServiceTypes.includes(hit.service_type!)
+      );
+
+      if (fauxNativeConnectors.length > 0) {
+        // There are some illegal service_types in their native connectors
+        deprecations.push({
+          level: 'critical',
+          deprecationType: 'feature',
+          title: i18n.translate('xpack.enterpriseSearch.deprecations.fauxNativeConnector.title', {
+            defaultMessage: 'Connectors with `is_native: true` must be of supported service types',
+          }),
+          message: i18n.translate(
+            'xpack.enterpriseSearch.deprecations.fauxNativeConnector.message',
+            {
+              values: { serviceTypes: nativeTypesStr },
+              defaultMessage:
+                'Only specific service types are supported for Elastic-managed connectors. ' +
+                'Eligible service types for Elastic-managed connectors are: {serviceTypes}' +
```

Comment:
I would suggest, if don't have time to prettify the formatting here, at least putting the list in its own separate paragraph to alleviate the wall of text somewhat.

You could make the list be the second paragraph, so you provide all the intro information together.

For example:

```markdown

Elastic-managed connectors don't support all service types.

Unsupported service types must be converted to Connector Clients before upgrading. This is a lossless operation, and can be attempted with "quick resolve". Alternatively, deleting these connectors with mismatched service types will also unblock your upgrade

The following service types are supported for Elastic-managed connectors:

{serviceTypes}
```

Reply:
++ on using paragraphs

@seanstory we're listing the service types twice in a message that is already long, do you think we could consolidate?

Reply:
++ made same [suggestion](https://github.com/elastic/kibana/pull/205500#discussion_r1906898087) about removing second instance of list